pkg/taskutil/taskutil.go

```diff
@@ -45,14 +47,23 @@ func NewTask(ctx context.Context, client *containerd.Client, container container
 		if err != nil {
 			return nil, err
 		}
-		ioCreator = cio.LogURI(u)
+		//using containerd runtime v2 logging plugin mode to handle task log entries
+		q := u.Query()
+		fn, err := loggingpkg.GetLoggerFunc(q.Get(loggingpkg.MagicArgv1))
+		if err != nil {
+			return nil, err
+		}
+		logging.Run(fn)
```

Comment:
This function can only be called from the logging plugin

Reply:
got it